writer.go

```diff
@@ -19,7 +19,25 @@ type Writer struct {
 	framer    Framer
 	formatter Formatter
 
-	conn serverConn
+	conn          serverConn
+	needReconnect bool
+}
+
+// Poll the socket to find out if/when it closes
+func (w *Writer) poll() {
+	for {
+		if w.conn == nil {
+			w.needReconnect = true
+			return
+		}
+
+		b := make([]byte, 1)
+		_, err := w.conn.read(b)
```

Comment:
My understanding is that reading from the socket is a blocking operation, so this would only be looping fast while we're actually receiving data.